client/src/components/FormField/RegisteredMFAMethodListField/AccountResetUI.js

```diff
@@ -0,0 +1,167 @@
+          onClick={this.handleSendReset}
+        >
+          {
+            i18n._t(
+              'MultiFactorAuthentication.ACCOUNT_RESET_ACTION',
+              fallbacks['MultiFactorAuthentication.ACCOUNT_RESET_ACTION']
+            )
+          }
+        </button>
+      </p>
+    );
+  }
+
+  renderSending() {
+    const { ss: { i18n } } = window;
+
+    return (
+      <p className="account-reset-action account-reset-action--sending">
+        <span className="account-reset-action__icon">
+          <LoadingIndicator size="32px" />
+        </span>
+        <span className="account-reset-action__message">
+          {
+            i18n._t(
+              'MultiFactorAuthentication.ACCOUNT_RESET_SENDING',
+              fallbacks['MultiFactorAuthentication.ACCOUNT_RESET_SENDING']
+            )
+          }
+        </span>
+      </p>
+    );
+  }
+
+  renderFailure() {
+    const { ss: { i18n } } = window;
+
+    return (
+      <p className="account-reset-action account-reset-action--failure">
+        <span className="account-reset-action__icon">
+          <CircleDash size="32px" />
+        </span>
+        <span className="account-reset-action__message">
+          {
+            i18n._t(
+              'MultiFactorAuthentication.ACCOUNT_RESET_SENDING',
+              fallbacks['MultiFactorAuthentication.ACCOUNT_RESET_SENDING_FAILURE']
+            )
+          }
+        </span>
+      </p>
+    );
+  }
+
+  renderSuccess() {
+    const { ss: { i18n } } = window;
+
+    return (
+      <p className="account-reset-action account-reset-action--success">
+        <span className="account-reset-action__icon">
+          <CircleTick size="32px" />
+        </span>
+        <span className="account-reset-action__message">
+          {
+            i18n._t(
+              'MultiFactorAuthentication.ACCOUNT_RESET_SENDING',
+              fallbacks['MultiFactorAuthentication.ACCOUNT_RESET_SENDING_SUCCESS']
+            )
+          }
+        </span>
+      </p>
+    );
+  }
+
+  render() {
+    const { ss: { i18n } } = window;
+    const { complete, failure, submitting } = this.state;
+
+    return (
+      <div className="account-reset">
+        <h5 className="account-reset__title">
+          {
+            i18n._t(
+              'MultiFactorAuthentication.ACCOUNT_RESET_TITLE',
+              fallbacks['MultiFactorAuthentication.ACCOUNT_RESET_TITLE']
+            )
+          }
+        </h5>
+
+        <p className="account-reset__description">
+          {
+            i18n._t(
+              'MultiFactorAuthentication.ACCOUNT_RESET_DESCRIPTION',
+              fallbacks['MultiFactorAuthentication.ACCOUNT_RESET_DESCRIPTION']
+            )
+          }
+        </p>
+
+        { !submitting && !complete && this.renderAction() }
+
+        { submitting && this.renderSending() }
+
+        { !submitting && complete && failure && this.renderFailure() }
+
+        { !submitting && complete && !failure && this.renderSuccess() }
```

Comment:
I think you could wrap these up into other methods to make it look a bit tidier, e.g.

```jsx
{ this.renderAction() }
{ this.renderStatusMessage() }
```

Then have each method handle its own display logic e.g.

```jsx
renderStatusMessage() {
  const { submitting, complete, failure } = this.state;
  if (submitting) {
     return this.renderSending();
  }
  if (failure) {
    return this.renderFailure();
  }
  return this.renderSuccess();
}
```

Reply:
I think we're approaching render method soup on this component (which to be fair probably means I should have child components...), but I get that the grouped checks are kind of hard to parse. I've refactored as requested.